File: C++/836-Rectangle_Overlap.cpp
// Runtime: 4 ms, faster than 49.45% of C++ online submissions for Rectangle Overlap.
// Memory Usage: 8.1 MB, less than 100.00% of C++ online submissions for Rectangle Overlap.
class Solution {
public:
    bool isRectangleOverlap(vector<int>& rec1, vector<int>& rec2) {
        return rec1[0] < rec2[2] && rec2[0] < rec1[2] && rec1[1] < rec2[3] && rec2[1] < rec1[3];
    }
};

File: Python3.x/836-Rectangle_Overlap.py
# Runtime: 32 ms, faster than 89.53% of Python3 online submissions for Rectangle Overlap.
# Memory Usage: 13.9 MB, less than 8.33% of Python3 online submissions for Rectangle Overlap.
class Solution:
    def isRectangleOverlap(self, rec1: List[int], rec2: List[int]) -> bool:
        return rec1[0] < rec2[2] and rec2[0] < rec1[2] and rec1[1] < rec2[3] and rec2[1] < rec1[3]


"""
(left1, right1), (left2, right2)
Meet the requirements of the topic  Equivalent to :
    left1 < x < right1 && left2 < x < right2
    left1 < x < right2 && left2 < x < right1
    left1 < right2 && left2 < right1
"""
